OpenEdXMobile/src/main/java/org/edx/mobile/module/db/impl/DbHelper.java

```diff
@@ -77,73 +80,86 @@ private void createAssessmentTable(SQLiteDatabase db){
+
                             while (cursor.moveToNext()) {
-                                final String id = cursor.getString(0);
-                                final String username = cursor.getString(1);
-                                final String filePath = cursor.getString(2);
+                                final String id = cursor.getString(idIndex);
+                                final String username = cursor.getString(usernameIndex);
+                                final String filePath = cursor.getString(filePathIndex);
                                 final String hashedUsername = Sha1Util.SHA1(username);
-                                final String newFilePath = filePath.replace(
-                                        appExternalDir.getAbsolutePath() + "/" + username,
-                                        appExternalDir.getAbsolutePath() + "/"
-                                                + AppConstants.Directories.VIDEOS + "/"
-                                                + hashedUsername);
-
-                                // First update the directory name along with its whole path
-                                final File previousDir = new File(appExternalDir, username);
+                                final String newFilePath = filePath.replaceFirst(
+                                        "^" + TextUtils.join("/", Arrays.<CharSequence>asList(
+                                                previousAppDirPath, username)),
+                                        TextUtils.join("/", Arrays.<CharSequence>asList(
+                                                externalAppDir.getAbsolutePath(),
+                                                AppConstants.Directories.VIDEOS,
+                                                hashedUsername)).toString());
+
+                                // First update the name and path of the videos directory
+                                final File previousDir = new File(externalAppDir, username);
                                 if (previousDir.exists()) {
-                                    final File newDir = new File(appExternalDir,
-                                            AppConstants.Directories.VIDEOS + "/" + hashedUsername);
-                                    newDir.mkdirs();
-                                    previousDir.renameTo(newDir);
+                                    final File newDir = new File(externalAppDir,
+                                            TextUtils.join("/", Arrays.<CharSequence>asList(
+                                                    AppConstants.Directories.VIDEOS,
+                                                    hashedUsername)).toString());
+                                    if (!((newDir.mkdirs() || newDir.exists()) &&
+                                            previousDir.renameTo(newDir))) {
+                                        continue;
```

Comment:
Hmm, what should we do with the database entry if this situation is somehow encountered. Maybe just delete the entry here?